mesonbuild/interpreter.py

```diff
@@ -2410,15 +2410,21 @@ def string_method_call(self, obj, method_name, args):
                 return obj.split(s)
             else:
                 return obj.split()
-        elif method_name == 'startswith' or method_name == 'contains' or method_name == 'endswith':
+        elif method_name == 'startswith' or method_name == 'contains' or method_name == 'endswith' or method_name == 'matches':
             s = posargs[0]
             if not isinstance(s, str):
                 raise InterpreterException('Argument must be a string.')
             if method_name == 'startswith':
                 return obj.startswith(s)
             elif method_name == 'contains':
                 return obj.find(s) >= 0
-            return obj.endswith(s)
+            elif method_name == 'endswith':
+                return obj.endswith(s)
+            elif method_name == 'matches':
+                for glob_pattern in s.split('|'):
```

Comment:
The `a|b|c` thing is also what's used by shell/configure case/esac, so it seems more natural to me and people might be more familiar with it?